wrapping/itkBlockMatchingMultiResolutionImageRegistrationMethod.wrap

```diff
@@ -0,0 +1,61 @@
+set(WRAPPER_AUTO_INCLUDE_HEADERS OFF)
+itk_wrap_include("itkPoint.h")
+itk_wrap_include("itkSmartPointer.h")
+itk_wrap_include("itkImage.h")
+itk_wrap_include("itkImageRegion.h")
+
+# Wrap base classes for itk::BlockMatching::MetricImageToDisplacementCalculator public members
+itk_wrap_class("itk::Image" POINTER)
+  foreach(t ${WRAP_ITK_REAL})
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      itk_wrap_template("P${ITKM_${t}}${d}${d}" "itk::Point<${ITKT_${t}},${d}>, ${d}")
+      itk_wrap_template("SPI${ITKM_${t}}${d}${d}" "itk::SmartPointer<itk::Image<${ITKT_${t}},${d}>>, ${d}")
+    endforeach()
+  endforeach()
+itk_end_wrap_class()
+
+itk_wrap_include("itkBlockMatchingMetricImageToDisplacementCalculator.h")
+itk_wrap_class("itk::BlockMatching::MetricImageToDisplacementCalculator" POINTER)
+  foreach(t ${WRAP_ITK_REAL})
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      itk_wrap_template("I${ITKM_${t}}${d}I${ITKM_V${t}${d}}${d}"
+                        "itk::Image<${ITKT_${t}},${d}>, itk::Image<${ITKT_V${t}${d}},${d}>")
+    endforeach()
+  endforeach()
+itk_end_wrap_class()
+
+# Wrap class hierarchy
+itk_wrap_include("itkImageToImageFilter.h")
+itk_wrap_class("itk::ImageToImageFilter" POINTER)
+  foreach(t ${WRAP_ITK_REAL})
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      itk_wrap_template("IIR${d}${d}I${ITKM_V${t}${d}}${d}"
+                        "itk::Image<itk::ImageRegion<${d}>,${d}>, itk::Image<${ITKT_V${t}${d}},${d}>")
+    endforeach()
+  endforeach()
+itk_end_wrap_class()
+
+itk_wrap_include("itkBlockMatchingImageRegistrationMethod.h")
+itk_wrap_class("itk::BlockMatching::ImageRegistrationMethod" POINTER)
```

Comment:
ITK implements an `ImageRegistrationMethod` and a `MultiResolutionImageRegistrationMethod` class. Instantiating these block matching registration classes from the Ultrasound submodule such as `itk.Ultrasound.ImageRegistrationMethod` returns the following warnings:
> Namespace already has a value for ImageRegistrationMethod, which is not an itkTemplate instance for class itk::BlockMatching::ImageRegistrationMethod. Overwriting old value.
Namespace already has a value for MultiResolutionImageRegistrationMethod, which is not an itkTemplate instance for class itk::BlockMatching::MultiResolutionImageRegistrationMethod. Overwriting old value.
<itkTemplate itk::BlockMatching::ImageRegistrationMethod>

Is there a better way to wrap so that we avoid these overrides? It sounds like we're already following the namespace for namespaces in [SWIG documentation](http://www.swig.org/Doc4.0/Python.html#Python_nn25):
>If you have more than one namespace and your want to keep their symbols separate, consider wrapping them as separate SWIG modules. For example, make the module name the same as the namespace and create extension modules for each namespace separately.

Reply:
I think we may need a unique class name. For example `itk::BlockMatching::ImageRegistrationMethod` -> `itk::BlockMatching::BlockMatchingImageRegistrationMethod`.

Reply:
Would this change have to happen directly for each `BlockMatching` class name with a conflict? Or can we specify an alias in the `.wrap` file?

Reply:
This would have to happen for each `BlockMatching` class name with a conflict, although I do not think there are many.

We want a unique name so the classes can be referenced uniquely with `itk.ClassName`.